include/engine/datafacade/shared_datafacade.hpp

```diff
@@ -534,22 +535,56 @@ class SharedDataFacade final : public BaseDataFacade
     }
 
     virtual void
-    GetUncompressedWeights(const EdgeID id,
-                           std::vector<EdgeWeight> &result_weights) const override final
+    GetUncompressedReverseGeometry(const EdgeID id,
+                                   std::vector<NodeID> &result_nodes) const override final
     {
-        const unsigned begin = m_geometry_indices.at(id);
+        const signed begin = m_geometry_indices.at(id) - 1;
+        const signed end = m_geometry_indices.at(id + 1) - 2;
+
+        result_nodes.clear();
```

Comment:
also, all of this looks like it is duplicated code, simply based on which datafacade we use? Doesn't this scream for a template function somewhere else that wraps the access for both datafacades, so we don't mess up two locations of code?